### ISSUE TITLE
Report Windows magnifier actions (changes in magnification levels etc)

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#15243]
### Is your feature request related to a problem? Please describe.

1. Press Windows+plus to turn on Windows magnifier
2. Press Windows+plus again to increase magnification
3. Press Windows-minus to decrease magnification
4. Press control+alt+i to invert colours
5. Press Windows+escape to exit Windows magnifier

Full list of keystrokes here: https://support.microsoft.com/en-us/windows/magnifier-keyboard-shortcuts-and-touch-gestures-6d7c3095-75ec-258f-6f5b-3b0bc19a18e7

Currently, NVDA does not report anything when these commands are executed.

### Describe the solution you'd like

NVDA should report each of those functions. E.G.:
1. "Windows Magnifier, enabled, full screen, 200% magnification."
2. "300% Magnification"
3. "200% Magnification"
4. "Invert colours enabled"
5. "Exiting Magnifier"

### Additional context
Tested just now on Windows 11 (64-bit) Version: 22H2, Build: 22621.1992  with NVDA 2023.2 Beta 1, but I don't think these changes were ever reported.

**Comment by Qchristensen** [comment]
Note: There is an add-on which reports some of these functions: https://addons.nvda-project.org/addons/winMag.en.html

**Comment by CyrilleB79** [comment]
Hi @Qchristensen

I am the author of Windows Magnifier add-on and may accept to integrate parts of it in NVDA if it is NVAccess' choice.

However this request raise some questions:

1. The shortcuts of Windows Magnifier have never been announced by NVDA, and probably neither by Narrator. Is there any specific reason why you have opened this issue today?

2. The add-on allows user to opt in or not (by installing the add-on or not). And when the add-on is installed it can be configured to report or not almost each shortcut separately. I imagine that in NVDA such granularity in the configuration would not be suitable, i.e. the result of all Magnifier shortcuts would be reported. I wonder if there are Windows Magnifier user who are not interested in having or even disturbed by shortcut reported vocally.

Cc @k-kolev1985 and @nishimotz who may be interested in this topic and may want to share their opinion.

**Comment by Qchristensen** [comment]
Hi Cyrille,

I actually completely forgot about the add-on when I first wrote the issue - so the issue is in no way a complaint about your add-on or its features.  (Although it's still arguably a valid ask).  I wrote it up following a user request.  The fact that there is an add-on potentially lowers the priority - after all there is currently a solution where for another bug there may not be. 

To your specific questions:

1) You're right, I don't think anything has changed in this regard - so it's not that this issue is suddenly more pressing, it was just that someone asked :)

2) Given the potential user base for this feature (ie those who use NVDA AND Magnifier, so are likely at least partially sighted), then yes, at the very least the ability to enable / disable reporting at all would be worthwhile.  Whether that is a simple on / off toggle, or more granular like the add-on I'm not sure - perhaps you would be in a good position to advise on that, and how your users currently use it?

**Comment by CyrilleB79** [comment]
> 2. Given the potential user base for this feature (ie those who use NVDA AND Magnifier, so are likely at least partially sighted), then yes, at the very least the ability to enable / disable reporting at all would be worthwhile.  Whether that is a simple on / off toggle, or more granular like the add-on I'm not sure - perhaps you would be in a good position to advise on that, and how your users currently use it?

No, I have no feedback on which option the add-on's users actually use. I think that I hade split the possibility to report actions in various options because:
* I have considered that some users may see easily the result of some actions (e.g. color inversion) but not other ones (e.g. modifying magnification level).
* Some magnifier shortcuts have been removed in recent versions of windows, e.g. alt+shift+arrow to resize the lense or docked window: thus the user may enable or disable reporting depending on if the shortcut actually belongs to Magnifier in his/her version of Windows
* The add-on may also report the location of the view when panning with control+alt+arrows; not all user want to have a feedback for each press to move the view.

I think that the number of options is OK in the add-on for someone who has opted in. But in NVDA the GUI should not be cluttered with so many options. I think that at most one option must be implemented (if any). This option should enable/disable reporting for the following actions: start/quit Magnifier, modify magnification level, color inversion, changing magnification type (fullscreen/lens/docked) as well as lens/docked window resizing (if possible). I think that reporting the location of the view when moving it (control+alt+arrow) is not suitable for an option in NVDA.

This is my opinion if a request is made to integrate reporting of Magnifier's commands in NVDA. But I am not very convinced that it would be a good idea. If we include Magnifier's command reporting, there would be many other add-on's feature which are more useful and asked more often which would deserve to be integrated in NVDA, e.g. report copy/paste operations.

**Comment by Adriani90** [comment]
@msftedad can you please report this to the development team?
I think this can be implemented on Microsoft side via UIA notification event or so, and a separate add-on for NVDA would be obsolete in that case.

**Comment by msftedad** [comment]
@Adriani90, We have forwarded this issue to development team. We will provide you with the tracking id once we receive update from them. Thanks
